src/java.base/share/classes/java/lang/annotation/ElementType.java

```diff
@@ -26,106 +26,213 @@
+ * <li>a non-void method (treated as if it precedes the method's
+ *     return type)
+ * <li>a constructor (treated as if it modifies the constructed type,
+ *     even though this is not technically a type context)
+ * </ul>
+ *
+ * <p>In general, a library method for reading declaration annotations
+ * (like {@link java.lang.reflect.Field#getAnnotations}) will not
+ * return type-use annotations found in the same location, and
+ * vice-versa.
+ *
+ * <p>An annotation interface may specify both {@link #TYPE_USE} and
+ * declaration targets, and thereby be fully usable as either kind.
+ * When an annotation of this type appears in one of the six ambiguous
+ * contexts just listed, it functions as <em>both</em> a declaration
+ * annotation and a type-use annotation at the same time. The results
+ * may be counterintuitive in two cases: when the variable type or
+ * method return type is an inner type or an array type. In these
+ * cases, the declaration annotation applies to the "entire"
+ * declaration, yet the type-use annotation applies more narrowly to
+ * the <em>outer type</em> or to the <em>component type</em> of the
+ * array.
  *
  * @author  Joshua Bloch
  * @since 1.5
- * @jls 9.6.4.1 @Target
- * @jls 4.1 The Kinds of Types and Values
+ * @jls 9.7.4 Where Annotations May Appear
  */
 public enum ElementType {
-    /** Class, interface (including annotation interface), enum, or record
-     * declaration */
+
+    /** 
+     * The declaration of a named class or interface. Classes without
+     * names, such as an anonymous class or an enum constant with a
+     * class body, cannot be annotated.
+     *
+     * <p><b>Terminology note:</b> despite this constant's name, an
+     * annotation applied to a class declaration is not a "type
+     * annotation". That phrase is only ever used as an abbreviation
+     * of "type-use annotation", which is supported by {@link
+     * #TYPE_USE}.
+     */
     TYPE,
 
-    /** Field declaration (includes enum constants) */
+    /**
+     * The declaration of a field (including of an enum constant).
+     *
+     * <p>Any annotation valid for a field declaration may also appear
+     * on the declaration of a record component, and is automatically
+     * copied to the private field of the same name that is generated
+     * during compilation.
+     */
     FIELD,
 
-    /** Method declaration */
+    /**
+     * The declaration of a method (including of an element of an
+     * annotation interface).
+     *
+     * <p>Any annotation valid for a method declaration may also appear
+     * on the declaration of a record component, and is automatically
+     * copied to the accessor method of the same name if one is
+     * generated during compilation.
+     */
     METHOD,
 
-    /** Formal parameter declaration */
+    /** 
+     * The declaration of a formal parameter (of a method, constructor,
+     * or lambda expression), exception parameter, or receiver
```

Comment:
Receiver parameters are type-use only; they were only added in Java 8. JLS 9.7.4:

> **It is a compile-time error if an annotation of interface A is syntactically a modifier for:**
> - [...]
> - a receiver parameter, but A is not applicable in type contexts.

Reply:
Good catch!